api/types/oidc.go

```diff
@@ -108,34 +108,17 @@ func NewOIDCConnector(name string, spec OIDCConnectorSpecV2) OIDCConnector {
 	}
 }
 
-// OIDCConnectorV2 is version 1 resource spec for OIDC connector
-type OIDCConnectorV2 struct {
-	// Kind is a resource kind
-	Kind string `json:"kind"`
-	// SubKind is a resource sub kind
-	SubKind string `json:"sub_kind,omitempty"`
-	// Version is version
-	Version string `json:"version"`
-	// Metadata is connector metadata
-	Metadata Metadata `json:"metadata"`
-	// Spec contains connector specification
-	Spec OIDCConnectorSpecV2 `json:"spec"`
-}
-
 // SetPrompt sets OIDC prompt value
 func (o *OIDCConnectorV2) SetPrompt(p string) {
-	o.Spec.Prompt = &p
+	o.Spec.Prompt = p
 }
 
 // GetPrompt returns OIDC prompt value,
 // * if not set, in this case defaults to select_account for backwards compatibility
 // * set to empty string, in this case it will be omitted
 // * and any non empty value, passed as is
 func (o *OIDCConnectorV2) GetPrompt() string {
-	if o.Spec.Prompt == nil {
-		return constants.OIDCPromptSelectAccount
-	}
-	return *o.Spec.Prompt
```

Comment:
@klizhentas Does this OIDC prompt still need to default to "select_account" for backwards compatibility, or can this be safely removed? I saw this was added around Teleport 4.2.

Reply:
This field isn't shown in example template when adding an OIDC connector via the web UI, so it's quite unlikely that customers are actually setting it themselves. I should think we need to keep the default.

```yaml
#
# Example resource for an OIDC connector
# We recommend using oidc for G Suite, Auth 0 and Keyclock
#
kind: oidc
version: v2
metadata:
  name: new_oidc_connector
spec:
  redirect_url: "https://<cluster-url>/v1/webapi/oidc/callback"
  client_id: <client id>
  # Connector display name that will be appended to the title of "Login with"
  # button on the cluster login screen so it will say "Login with Google".
  # Teleport will provide custom CSS for 'Google'.
  display: Google
  client_secret: <client secret>
  issuer_url: https://<issuer-url>
  scope: [<scope value>]
  claims_to_roles:
    - {claim: "hd", value: "example.com", roles: ["admin"]}
```

It looks like we only [mention it here](https://goteleport.com/docs/enterprise/sso/oidc/#optional-prompt) in the docs.